pkg/fakeoperator/internal/controllers/webhooks/db_webhook.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhooks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	adminv1beta1 "k8s.io/api/admission/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	fakev1alpha1 "github.com/kubermatic/kubecarrier/pkg/apis/fake/v1alpha1"
+	"github.com/kubermatic/kubecarrier/pkg/internal/util/webhook"
+)
+
+const defaultPassword = "password"
+
+// DBWebhookHandler handles mutating/validating of DBs.
+type DBWebhookHandler struct {
+	decoder *admission.Decoder
+	Log     logr.Logger
+	Scheme  *runtime.Scheme
+	Client  client.Client
+}
+
+var _ admission.Handler = (*DBWebhookHandler)(nil)
+
+// +kubebuilder:webhook:path=/mutate-fake-kubecarrier-io-v1alpha1-db,mutating=true,failurePolicy=fail,groups=fake.kubecarrier.io,resources=dbs,verbs=create;update;delete,versions=v1alpha1,name=mdb.kubecarrier.io
+
+// Handle is the function to handle create/update requests of DBs.
+func (r *DBWebhookHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	switch req.Operation {
+	case adminv1beta1.Create:
+		obj := &fakev1alpha1.DB{}
```

Comment:
could you also make `validateCreate` helper function to be consistent with `validateUpdate`/`validateDelete`?

Reply:
https://github.com/kubermatic/kubecarrier/blob/81fd633002ba1060361f4ab4885cf6a61d9e1cec/pkg/fakeoperator/internal/controllers/webhooks/db_webhook.go#L57-L59
we already have it

Reply:
oh ok, haven't noticed it....then why are you handing further work inside this function? the Defaulting should either be in `validateCreate` or even better in `defaultObject` webhook method. Just to keep it nice and tidy in the main Handle function.

Reply:
done